### ISSUE TITLE
Not finding device

### ISSUE THREAD

**Issue by r1plentz** [GSConnect/gnome-shell-extension-gsconnect#1036]
**Describe the bug**

GSConnect doesn't find my device (the 'Searching for devices...' text keeps on the screen undefinitely).

**Steps To Reproduce:**

1. Go to 'Mobile Devices'
2. Click on 'Turn On'
3. Click on 'Mobile Settings'
4. Click 'Refresh' (left-up corner)
5. See error

**Expected behavior**

After a while, the appearance of my mobile device name as seen in my working laptop.

**Screenshots**

![Screenshot from 2021-01-02 06-34-46](https://user-images.githubusercontent.com/63497545/103454625-bb9f1c80-4cc4-11eb-825f-44eea971af69.png)

**Support Log**

Please generate a support log ([Instructions](../wiki/Help#generate-support-log)) and paste any messages related to this issue between the two ``` lines below.

```

GSConnect: 44 (user)
GJS:       16403
Session:   x11
OS:        Ubuntu 20.04.1 LTS
--------------------------------------------------------------------------------
-- Logs begin at Mon 2020-10-12 20:56:24 -03, end at Sat 2021-01-02 06:58:58 -03. --
-- No entries --
```

**System Details (please complete the following information):**

 - **GSConnect version:** 44
   - **Installed from:** GitHub
   -  - **GNOME/Shell version:** 3.36
 - **Distro/Release:** Ubuntu 20.04

**GSConnect environment (if applicable):**

 - **Paired Device(s):** none (see 'Describe the bug')
 - **KDE Connect app version:** 1.15.1

**Additional Notes:**

I already tried to check the firewall solution. I'm currently using ufw and the ports were opened. The solution worked in my laptop spects fine, even if it was connecting before the firewall was enabled.

**Comment by daniellandau** [comment]
To clarify:

1. You have another machine where you can connect?
1. You can connect on this machine if you disable the firewall?

**Comment by r1plentz** [comment]
>     1. You have another machine where you can connect?
Yes, I have a laptop which also runs Ubuntu 20.04. The running version is 43, but it was bugging before I updated this weak (I thought the latest would fix it).

>     2. You can connect on this machine if you disable the firewall?
On my laptop, it works with or without firewall (when enable I have to allow the ports). On my desktop both options don't work.

**Comment by andyholmes** [comment]
Can you confirm you followed all the steps in [the Wiki](https://github.com/GSConnect/gnome-shell-extension-gsconnect/wiki/Help#connecting-an-android-device)? Can you ping your phone from your desktop? If your laptop uses WiFi and your desktop is wired, is your router isolating the WiFi from the wired?

**Comment by DaddyWarlock** [comment]
I'm having a similar, if not the same, problem.
GNOME 3.38 on Ubuntu 21.04 with GSConnect version 45
Samsung Galaxy S20 FE 5G running KDE Connect 1.17
I can ping the phone from my laptop, but the phone never sees any other devices on the network.  Any ideas?  Should I open a new issue on this?

**Comment by andyholmes** [comment]
Can you confirm you followed all the steps in [the Wiki](https://github.com/GSConnect/gnome-shell-extension-gsconnect/wiki/Help#connecting-an-android-device), including adding devices by IP?

**Comment by DaddyWarlock** [comment]
On Mon, 2021-07-05 at 08:28 -0700, Andy Holmes wrote:
> Can you confirm you followed all the steps in the Wiki, including
> adding devices by IP?

I have installed the laptop by IP on the phone to no avail, but I can't
even get GSConnect to start.  I believe I've followed all the steps in
the Wiki, yes.  Until I can get GSConnect to start properly, I don't
see how this is going to work.

**Comment by andyholmes** [comment]
> Until I can get GSConnect to start properly, I don't see how this is going to work.

Heh, yep. Your problem is definitely GSConnect not starting, not a connection problem.

You'll need logs from either GNOME Logs or `journalctl` to diagnose the problem.

**Comment by DaddyWarlock** [comment]
On Mon, 2021-07-05 at 08:38 -0700, Andy Holmes wrote:
> > Until I can get GSConnect to start properly, I don't see how this
> > is going to work.
> Heh, yep. Your problem is definitely GSConnect not starting, not a
> connection problem.
> You'll need logs from either GNOME Logs or journalctl to diagnose the
> problem.
Okay, this is a definite dumbass question, but where do I find the
GNOME logs?  I checked /var/log and there's no GNOME directory or file
there.  Sorry to be so oblivious!

**Comment by andyholmes** [comment]
That's okay. [GNOME Logs](https://flathub.org/apps/details/org.gnome.Logs) is just simple GUI for browsing logs. Otherwise you could try something like `journalctl | grep gsconnect` or `journalctl | grep GSConnect`.

**Comment by DaddyWarlock** [comment]
On Mon, 2021-07-05 at 08:38 -0700, Andy Holmes wrote:
> > Until I can get GSConnect to start properly, I don't see how this
> > is going to work.
> Heh, yep. Your problem is definitely GSConnect not starting, not a
> connection problem.

Okay, I rebooted and now GSConnect starts.  Problem is it appears to be
in a loop searching for devices.  If it's not one thing, it's another.

This wouldn't have anything to do with both devices running Wireguard
VPN, would it?

**Comment by DaddyWarlock** [comment]
Following up to my previous comment.  I turned off the VPN on both the phone and the laptop and they saw each other.  I was able to pair them and things are working now.  But is there a way to do it with the VPN still running?

**Comment by andyholmes** [comment]
Yup, the Wiki has links to [KDE Connect's](https://userbase.kde.org/KDEConnect#Running_KDE_Connect_over_OpenVPN) instructions.

**Comment by Greifent** [comment]
Hi, I'm having a very similar issue: the computer cannot find the phone;
here there is the log from GSconnect
```
GSConnect: 45 (user)
GJS:       16702
Session:   x11
OS:        Pop!_OS 21.04
--------------------------------------------------------------------------------
-- Journal begins at Wed 2021-02-03 09:59:27 CET, ends at Tue 2021-07-06 20:27:05 CEST. --
Jul 06 20:27:05 org.gnome.Shell.Extensions.GSConnect[33350]: [/service/backends/lan.js:broadcast:413]: "Broadcasting to LAN"
```
Running ``journalctl | grep gsconnect`` I get:
 ``Jul 05 22:46:20 pop-os gjs[3865]: gsconnect: Fatal IO error 11 (Resource temporarily unavailable) on X server :1.``
I tried adding by IP but nothing

**Comment by DaddyWarlock** [comment]
On Tue, 2021-07-06 at 11:32 -0700, Riccardo Vanin wrote:
> Hi, I'm having a very similar issue: the computer cannot find the
> phone;
> here there is the log from GSconnect
> GSConnect: 45 (user)
> GJS:       16702
> Session:   x11
> OS:        Pop!_OS 21.04
> ---------------------------------------------------------------------
> -----------
> -- Journal begins at Wed 2021-02-03 09:59:27 CET, ends at Tue 2021-
> 07-06 20:27:05 CEST. --
> Jul 06 20:27:05 org.gnome.Shell.Extensions.GSConnect[33350]:
> [/service/backends/lan.js:broadcast:413]: "Broadcasting to LAN"
> Running journalctl | grep gsconnect I get:
> Jul 05 22:46:20 pop-os gjs[3865]: gsconnect: Fatal IO error 11
> (Resource temporarily unavailable) on X server :1.
> I tried adding by IP but nothing

You've probably done this, but did you open the ports for GSConnect in
the firewall?

**Comment by DaddyWarlock** [comment]
I've been doing a little further playing around, and I find the main issue is the Wireguard VPN on the Android.  When it's turned off, I can connect to the laptop, even if the laptop's Wireguard connection is turned on.  Any one have any ideas?

**Comment by daniellandau** [comment]
Android's VPN handling is goofy. I haven't paid attention to GSConnect, but I get only a partial connection on jitsi when I have a wireguard connection open 🤷

**Comment by Greifent** [comment]
>You've probably done this, but did you open the ports for GSConnect in
the firewall?
It used to work before, I'll take a look anyway

**Comment by DaddyWarlock** [comment]
On Tue, 2021-07-06 at 22:51 -0700, Daniel Landau wrote:
> Android's VPN handling is goofy. I haven't paid attention to
> GSConnect, but I get only a partial connection on jitsi when I have a
> wireguard connection open 🤷

After some additional messing about, I've found I can add the Wireguard
IP address directly and everything works.  Thanks to everyone who
helped with this issue!

**Comment by mnovozhylov** [comment]
I have similar issue, but in my case I found the following in the logs:
```
Jul 08 08:30:24 up org.gnome.Shell.Extensions.GSConnect[7006]: [/service/backends/lan.js:broadcast:413]: "Broadcasting to LAN"
Jul 08 08:30:24 up org.gnome.Shell.Extensions.GSConnect[7006]: [/service/backends/lan.js:accept:779]: "lan://192.168.2.102:1716 (344e2794-d51e-41f2-9ab4-846b1085dac2)"
Jul 08 08:30:25 up org.gnome.Shell.Extensions.GSConnect[7006]: [/service/backends/lan.js:close:820]: "lan://192.168.2.102:1716 (344e2794-d51e-41f2-9ab4-846b1085dac2)"
Jul 08 08:30:25 up org.gnome.Shell.Extensions.GSConnect[7006]: [/service/backends/lan.js:_handshake/</<:595]: Peer sent fatal TLS alert: Unknown certificate
                                                               _handshake/</<@/home/m/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/backends/lan.js:595:44
```
I'd appreciate any ideas how to fix this

**Comment by andyholmes** [comment]
Closing since the issue is vague (probably a VPN problem anyways) and attracting unrelated issues.

**Comment by iamYashSinha** [comment]
I'd suggest, changing the firewall settings and setting the **incoming: Allow, Outgoing: Deny**, btw I'm using Ubuntu 22.04.3 LTS

**Comment by volek75** [comment]
Doesn't work on gnome 48. Please complete the add-on for this version.
